File: r/hello_world.R
# Hello World example for GitHub Copilot in R
# Try typing comments like "# Function to print hello world"
# and see how Copilot suggests the implementation

# Function to print a greeting message
print_greeting <- function() {
  print("Hello, GitHub Copilot!")
}

# Function to get user name from input
get_user_name <- function() {
  cat("Please enter your name: ")
  name <- readline()
  return(name)
}

# Function to greet a specific user
greet_user <- function(name) {
  cat(paste("Hello,", name, "! Welcome to GitHub Copilot examples.\n"))
}

# Function to display current date and time
display_current_datetime <- function() {
  current_time <- Sys.time()
  cat("Current date and time:", format(current_time, "%Y-%m-%d %H:%M:%S"), "\n")
}

# Function to generate random R facts
get_random_r_fact <- function() {
  facts <- c(
    "R was created by Ross Ihaka and Robert Gentleman",
    "R is excellent for statistical computing and graphics",
    "R has thousands of packages available on CRAN",
    "GitHub Copilot can help you write R code faster",
    "R's vectorized operations make it very efficient"
  )
  
  # Return a random fact
  return(sample(facts, 1))
}

# Function to create and analyze a simple vector
analyze_numbers <- function() {
  # Create a vector of numbers
  numbers <- c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
  
  cat("Numbers:", paste(numbers, collapse = ", "), "\n")
  cat("Sum:", sum(numbers), "\n")
  cat("Mean:", mean(numbers), "\n")
  cat("Median:", median(numbers), "\n")
  cat("Standard deviation:", sd(numbers), "\n")
  
  return(numbers)
}

# Function to create a simple data frame
create_sample_data <- function() {
  # Create sample data
  data <- data.frame(
    name = c("Alice", "Bob", "Carol", "David", "Eve"),
    age = c(25, 30, 28, 35, 22),
    score = c(85, 92, 78, 88, 95),
    city = c("New York", "London", "Paris", "Tokyo", "Sydney")
  )
  
  cat("Sample data created:\n")
  print(data)
  return(data)
}

# Function to perform basic data analysis
analyze_sample_data <- function(data) {
  cat("\nData Analysis Summary:\n")
  cat("Number of records:", nrow(data), "\n")
  cat("Average age:", mean(data$age), "\n")
  cat("Average score:", mean(data$score), "\n")
  cat("Age range:", min(data$age), "to", max(data$age), "\n")
  cat("Score range:", min(data$score), "to", max(data$score), "\n")
  
  # Find highest scorer
  highest_scorer <- data[which.max(data$score), ]
  cat("Highest scorer:", highest_scorer$name, "with score", highest_scorer$score, "\n")
}

# Main function to demonstrate R with GitHub Copilot
main <- function() {
  cat("R GitHub Copilot Demo\n")
  cat(rep("=", 30), "\n", sep = "")
  
  # Print greeting
  print_greeting()
  
  # Get user name and greet them
  # Uncomment the following lines for interactive mode
  # user_name <- get_user_name()
  # greet_user(user_name)
  
  # For demo purposes, use a default name
  user_name <- "R Programmer"
  greet_user(user_name)
  
  # Display current date and time
  display_current_datetime()
  
  # Show a random R fact
  cat("\nRandom R fact:", get_random_r_fact(), "\n\n")
  
  # Analyze numbers
  cat("Number Analysis:\n")
  numbers <- analyze_numbers()
  
  # Create and analyze sample data
  cat("\nData Frame Example:\n")
  sample_data <- create_sample_data()
  analyze_sample_data(sample_data)
  
  # Create a simple plot (if running in an environment that supports graphics)
  if (interactive()) {
    cat("\nCreating a simple plot...\n")
    plot(sample_data$age, sample_data$score, 
         main = "Age vs Score", 
         xlab = "Age", 
         ylab = "Score",
         pch = 19, 
         col = "blue")
    
    # Add a trend line
    lm_model <- lm(score ~ age, data = sample_data)
    abline(lm_model, col = "red", lwd = 2)
  }
  
  cat("\nDemo completed!\n")
}

# Run the main function
main()

File: java/HelloWorld.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Random;
import java.util.Scanner;

/**
 * Hello World example for GitHub Copilot in Java
 * Try typing comments like "// Method to print hello world"
 * and see how Copilot suggests the implementation
 */
public class HelloWorld {
    
    private static final Scanner scanner = new Scanner(System.in);
    private static final Random random = new Random();
    
    /**
     * Method to print a greeting message
     */
    public static void printGreeting() {
        System.out.println("Hello, GitHub Copilot!");
    }
    
    /**
     * Method to get user name from input
     */
    public static String getUserName() {
        System.out.print("Please enter your name: ");
        return scanner.nextLine();
    }
    
    /**
     * Method to greet a specific user
     */
    public static void greetUser(String name) {
        System.out.println("Hello, " + name + "! Welcome to GitHub Copilot examples.");
    }
    
    /**
     * Method to display current date and time
     */
    public static void displayCurrentDateTime() {
        LocalDateTime now = LocalDateTime.now();
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        System.out.println("Current date and time: " + now.format(formatter));
    }
    
    /**
     * Method to generate random Java facts
     */
    public static String getRandomJavaFact() {
        String[] facts = {
            "Java was originally called Oak",
            "Java is platform independent thanks to JVM",
            "Java follows 'Write Once, Run Anywhere' principle",
            "GitHub Copilot can help you write Java code faster",
            "Java is strongly typed and object-oriented"
        };
        
        return facts[random.nextInt(facts.length)];
    }
    
    /**
     * Method to demonstrate basic array operations
     */
    public static void demonstrateArrayOperations() {
        System.out.println("\nArray Operations Demo:");
        
        // Create an array of numbers
        int[] numbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        
        System.out.print("Numbers: ");
        for (int i = 0; i < numbers.length; i++) {
            System.out.print(numbers[i]);
            if (i < numbers.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println();
        
        // Calculate sum
        int sum = 0;
        for (int number : numbers) {
            sum += number;
        }
        System.out.println("Sum: " + sum);
        
        // Calculate average
        double average = (double) sum / numbers.length;
        System.out.println("Average: " + String.format("%.2f", average));
        
        // Find min and max
        int min = numbers[0];
        int max = numbers[0];
        for (int number : numbers) {
            if (number < min) min = number;
            if (number > max) max = number;
        }
        System.out.println("Min: " + min + ", Max: " + max);
    }
    
    /**
     * Method to demonstrate string operations
     */
    public static void demonstrateStringOperations() {
        System.out.println("\nString Operations Demo:");
        
        String text = "GitHub Copilot helps developers";
        
        System.out.println("Original text: " + text);
        System.out.println("Length: " + text.length());
        System.out.println("Uppercase: " + text.toUpperCase());
        System.out.println("Lowercase: " + text.toLowerCase());
        System.out.println("Contains 'Copilot': " + text.contains("Copilot"));
        System.out.println("Starts with 'GitHub': " + text.startsWith("GitHub"));
        System.out.println("Replace 'helps' with 'assists': " + text.replace("helps", "assists"));
        
        // Split the string
        String[] words = text.split(" ");
        System.out.print("Words: ");
        for (int i = 0; i < words.length; i++) {
            System.out.print(words[i]);
            if (i < words.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println();
    }
    
    /**
     * Method to create and display user information
     */
    public static void createUserInfo(String userName) {
        System.out.println("\nUser Information:");
        System.out.println("Name: " + userName);
        System.out.println("Timestamp: " + LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
        System.out.println("Favorite Language: Java");
        System.out.println("Using: GitHub Copilot");
        
        // Calculate name statistics
        System.out.println("Name length: " + userName.length());
        System.out.println("Name in uppercase: " + userName.toUpperCase());
        System.out.println("First character: " + (userName.isEmpty() ? "N/A" : userName.charAt(0)));
    }
    
    /**
     * Main method to demonstrate GitHub Copilot capabilities
     */
    public static void main(String[] args) {
        System.out.println("Java GitHub Copilot Demo");
        System.out.println("========================");
        
        // Print greeting
        printGreeting();
        
        // Get user name and greet them
        String userName = getUserName();
        greetUser(userName);
        
        // Display current date and time
        displayCurrentDateTime();
        
        // Show a random Java fact
        System.out.println("\nRandom Java fact: " + getRandomJavaFact());
        
        // Demonstrate array operations
        demonstrateArrayOperations();
        
        // Demonstrate string operations
        demonstrateStringOperations();
        
        // Create user information
        createUserInfo(userName);
        
        System.out.println("\nDemo completed!");
        System.out.println("Try writing your own methods using GitHub Copilot!");
        
        scanner.close();
    }
}